File: menagerie/bug-zoo/species/BZ-SHAPE-005-null-boundary-equivalence/csharp/fixed/provekit-annotations/harness/src/UserDirectory.cs
using System;

namespace BugZoo.CSharpNullBoundary.ProvekitAnnotations;

public static class UserDirectory
{
    //provekit:contract
    public static string Lookup(string? name)
    {
        if (name is null)
        {
            throw new ArgumentNullException(nameof(name), "name must be non-null");
        }

        return "user:" + name.ToUpperInvariant();
    }
}

File: menagerie/bug-zoo/species/BZ-SHAPE-005-null-boundary-equivalence/java/fixed/provekit-native/harness/src/main/java/zoo/UserDirectory.java
package zoo;

import com.provekit.contract.NotNull;

public final class UserDirectory {
    public String lookup(@NotNull String name) {
        if (name == null) {
            throw new IllegalArgumentException("name must be non-null");
        }
        return "user:" + name.toUpperCase();
    }
}
